src/main/java/org/candlepin/resource/RootResource.java

```diff
@@ -85,15 +115,21 @@ public RootResource(Configuration config) {
     protected List<Link> createLinks() {
         // Hidden resources will be omitted from the supported list we send to
         // the clients:
-        List<String> hideResources = Arrays.asList(
-            config.getString(ConfigProperties.HIDDEN_RESOURCES).split(" "));
+        Set<String> hideResources = Set.of(config.getString(ConfigProperties.HIDDEN_RESOURCES).split(" "));
+        Map<String, Link> links = new HashMap<>();
 
-        List<Link> newLinks = new LinkedList<>();
         for (Map.Entry<Object, String> entry : RESOURCE_CLASSES.entrySet()) {
-            add(resourceLink(entry.getKey(), entry.getValue()), hideResources,
-                newLinks);
+            Link link = this.resourceLink(entry.getKey(), entry.getValue());
+
+            if (hideResources.contains(link.getRel())) {
+                log.debug("Hiding supported resource: {}", link.getRel());
+                continue;
+            }
+
+            links.put(link.getRel(), link);
         }
-        return newLinks;
+
+        return new LinkedList<>(links.values()); // :/
```

Comment:
This probably should be fixed. I didn't intend for this to go out entirely as-is, since it's rather inefficient. It'd be better if we could just return a collection rather than a list so we don't need to build a new list just to output the correct type.

Reply:
Ack, Done.